lib/server.js

```diff
@@ -35,11 +36,14 @@ class UtapiServer {
     requestListener(req, res, router) {
         // disable nagle algorithm
         req.socket.setNoDelay();
+        const { query, path } = url.parse(req.url, true);
         const utapiRequest = new UtapiRequest()
             .setRequest(req)
             .setLog(this.logger.newRequestLogger())
             .setResponse(res)
-            .setDatastore(this.datastore);
+            .setDatastore(this.datastore)
+            .setRequestQuery(query)
```

Comment:
since you use url.parse(), do you still need decodeURIComponent in those functions ?